From 3ccb71eeca42dbcd5e4d00ae1877a489ae82598d Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Wed, 29 Dec 2021 16:04:54 +0800
Subject: [PATCH] Disable PIC explicitly for assembling

For some toolchain, this patch is needed if the fw_dynamic load
address is not equal to link address. However we've added an u-boot
patch that prevents fw_dynamic relocation by loading it to 0x0
rather than 0x1000000.

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index d6f097d..441518d 100644
--- a/Makefile
+++ b/Makefile
@@ -225,6 +225,7 @@ ASFLAGS		+=	-mcmodel=$(PLATFORM_RISCV_CODE_MODEL)
 ASFLAGS		+=	$(GENFLAGS)
 ASFLAGS		+=	$(platform-asflags-y)
 ASFLAGS		+=	$(firmware-asflags-y)
+ASFLAGS		+=	-fno-pic

 ARFLAGS		=	rcs

--
2.25.1

